molgenis-questionnaires/src/main/frontend/src/store/actions.js

```diff
@@ -83,21 +85,29 @@ const actions = {
     })
   },
 
-  'AUTO_SAVE_QUESTIONNAIRE' ({commit, state}: VuexContext, formData: Object) {
-    commit('INCREMENT_SAVING_QUEUE')
-    const updatedAttribute = Object.keys(formData).find(key => formData[key] !== state.formData[key]) || ''
-
-    const options = {
-      body: JSON.stringify(formData[updatedAttribute]),
-      method: 'PUT'
-    }
+  'AUTO_SAVE_QUESTIONNAIRE' ({commit, state}: VuexContext, payload: Object) {
+    const {formData, formState} = payload
 
-    return api.post(`/api/v1/${state.questionnaire.meta.name}/${state.questionnaireRowId}/${updatedAttribute}`, options).then(() => {
-      commit('SET_FORM_DATA', formData)
-    }, error => {
-      handleError(commit, error)
-    }).then(() => {
-      commit('DECREMENT_SAVING_QUEUE')
+    const updatedAttribute = Object.keys(formData).find(key => formData[key] !== state.formData[key]) || ''
+    commit('SET_FORM_DATA', formData)
+
+    Vue.nextTick(() => {
+      // TODO To refactor this code without the nextTick method we need to expose the validation
+      // TODO from the molgenis-ui-form to the interface of molgenis-ui-form
+      if (formState.$valid) {
+        const options = {
+          body: JSON.stringify(formData[updatedAttribute]),
+          method: 'PUT'
+        }
+
+        commit('INCREMENT_SAVING_QUEUE')
+
+        return api.post(`/api/v1/${state.questionnaire.meta.name}/${state.questionnaireRowId}/${updatedAttribute}`, options).catch((error) => {
```

Comment:
Please encode your URI path parts to prevent escaping issues:
```
'/api/v1/' + encodeURIComponent(${state.questionnaire.meta.name}) + '/' + encodeURIComponent(${state.questionnaireRowId}) + '/' + encodeURIComponent(${updatedAttribute})
```

Reply:
these values come from the server, seems like overkill to encode these

Reply:
add 'encodeURIComponent' to all params